extra/onnx_ops.py

```diff
@@ -207,7 +207,7 @@ def Not(x:Tensor): return Where((x==1), Tensor.zeros(*x.shape), Tensor.ones(*x.s
 
 def ConstantOfShape(input, value=0.0):
   shape = [int(x) for x in safe_numpy(input)]
-  return Tensor.ones(*shape) * value
+  return Tensor.ones(*shape, dtype=value.dtype) * value 
```

Comment:
Err, what if value is 0.0?

Reply:
@geohot  yeah I mentioned in a comment above:

The if was necessary when the value was 0.0 otherwise throws AssertionError: zeros not allowed in shape (0,) however it would be enough to just check the shape `if input.shape != (0,) 

I was wondering also if Tinygrad will support the behavior below then the if wouldn't be needed at all

```python
torch.ones(0) * 0.0 # realizes into tensor([])
Tensor.ones(0) * 0.0 # AssertionError: zeros not allowed in shape (0,)
```
Which one of the two should I follow?

Reply:
0.0.dtype